src/Dev/Deprecation.php

```diff
@@ -234,14 +234,22 @@ public static function get_enabled()
 
     public static function isEnabled(): bool
     {
+        // Return early if disabled
         if (Environment::hasEnv('SS_DEPRECATION_ENABLED')) {
-            $envVar = Environment::getEnv('SS_DEPRECATION_ENABLED');
-            return self::varAsBoolean($envVar);
+            if (!Environment::getEnv('SS_DEPRECATION_ENABLED')) {
+                return false;
+            }
+        } elseif (!static::$currentlyEnabled) {
+            // Static property is ignored if SS_DEPRECATION_ENABLED was set
+            return false;
         }
```

Comment:
I guess this could also be written like so:

```php
        if (Environment::hasEnv('SS_DEPRECATION_ENABLED') && !self::varAsBoolean(Environment::getEnv('SS_DEPRECATION_ENABLED'))) {
            return false;
        }
        // Static property is ignored if SS_DEPRECATION_ENABLED was set
        if (!Environment::hasEnv('SS_DEPRECATION_ENABLED') && !static::$currentlyEnabled) {
            return false;
        }
```
That doubles up the `hasENV` check but is easier to read. @emteknetnz @michalkleiner let me know which you'd prefer between the current code and this.

Reply:
I could also have a `$envEnabledIsSet = Environment::hasEnv('SS_DEPRECATION_ENABLED')` before the conditions I guess and just check against that variable in the two conditional statements

Reply:
This looks better and is basically what I had written up above, just doing 1, 3, 2&4, and it would be in a separate blocks as Steve suggested..

Reply:
Yep the separate blocks easier to understand

Reply:
Done